File: 1730-special-array-with-x-elements-greater-than-or-equal-x/special-array-with-x-elements-greater-than-or-equal-x.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var specialArray = function(nums) {
    nums.sort((a, b) => b - a);
    for (let i = 1; i <= nums.length; i++) {
        if (nums[i - 1] >= i && (i === nums.length || nums[i] < i)) {
            return i;
        }
    }
    return -1;
};

File: 1730-special-array-with-x-elements-greater-than-or-equal-x/special-array-with-x-elements-greater-than-or-equal-x.py
class Solution:
    def specialArray(self, nums: List[int]) -> int:
        nums.sort(reverse=True)
        for i in range(1, len(nums) + 1):
            if nums[i - 1] >= i and (i == len(nums) or nums[i] < i):
                return i
        return -1